pkg/network/controllers/vm.go

```diff
@@ -154,3 +123,27 @@ func (v *VMController) vmiInterfacesPatch(newVmiSpec *v1.VirtualMachineInstanceS
 
 	return err
 }
+
+func hasOrdinalSecondaryIfaces(
+	networks []v1.Network,
+	ifaceStatuses []v1.VirtualMachineInstanceNetworkInterface,
+) bool {
+	secondaryNets := vmispec.FilterMultusNonDefaultNetworks(networks)
+	if len(secondaryNets) == 0 {
+		return false
+	}
+
+	ifaceStatusesByName := vmispec.IndexInterfaceStatusByName(ifaceStatuses, nil)
+	for _, net := range secondaryNets {
+		ifaceStatus, ok := ifaceStatusesByName[net.Name]
+		if !ok || ifaceStatus.PodInterfaceName == "" {
+			continue
+		}
+
+		return namescheme.OrdinalSecondaryInterfaceName(ifaceStatus.PodInterfaceName)
```

Comment:
Wouldn't this mistakenly return false if the first secondaryNet is not in ordinal form?

Reply:
Currently, KubeVirt has:
- A hashed naming scheme (pod<hash>).
- A deprecated ordinal naming scheme (net1, net2, netN...)

A VM could have just one naming scheme.

For technical reasons (which I can elaborate on in case you are interested), VMs that had ordinal naming scheme - retained their naming scheme on migration (and upgrade).

There is an existing logic to prevent NIC hotunplug in VMs with ordinal naming scheme.
Since the function is named `hasOrdinalSecondaryIfaces` and the first secondary network does not have an ordinal form - it should return `false`.

Reply:
> A VM could have just one naming scheme.

Ah this was the missing part for me, thanks